File: NaviTest/NaviTest/PinkieNavigation/PinkieViewController.m
//
//  PinkieViewController.m
//  NaviTest
//
//  Created by jiazhiyi on 2019/4/9.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

#import "PinkieViewController.h"
#import "PinkieNavigationController.h"

@interface PinkieViewController ()

@end

@implementation PinkieViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
}

- (PinkieNavigationItem *)pinkieItem
{
    if (_pinkieItem == nil) {
        _pinkieItem = [[PinkieNavigationItem alloc] init];
    }
    return _pinkieItem;
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    PinkieNavigationController *nav = (PinkieNavigationController *)self.navigationController;
    PinkieNavigationView *navigationView = nav.navigationView;
    if (animated) {
        [navigationView addNavigationItem:self.pinkieItem];
        return;
    }
    
    [navigationView removeAllItems];
    CGFloat width = [UIScreen mainScreen].bounds.size.width;
    
    if (self.pinkieItem.navigationHidden) {
        CGRect rect = navigationView.frame;
        rect.origin.x = width;
        navigationView.frame = rect;
    } else {
        CGRect rect = navigationView.frame;
        rect.origin.x = 0;
        navigationView.frame = rect;
        [navigationView addNavigationItem:self.pinkieItem];
    }
}


@end

File: CustomNavigationBar/CustomNavigationBar/PinkieNavigation/PinkieViewController.swift
//
//  PinkieViewController.swift
//  CustomNavigationBar
//
//  Created by jiazhiyi on 2019/4/4.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

import UIKit

class PinkieViewController: UIViewController {
    
    var pinkieItem = PinkieNavigationItem()

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        let nav = self.navigationController as! PinkieNavigationController
        let navigationView = nav.navigationView!
        if animated {
            navigationView.addNavigationItem(item: self.pinkieItem)
            return
        }
        navigationView.removeAllItems()
        let width = UIScreen.main.bounds.size.width
        if self.pinkieItem.navigationHidden {
            var rect = navigationView.frame
            rect.origin.x = width
            navigationView.frame = rect
        } else {
            var rect = navigationView.frame
            rect.origin.x = 0
            navigationView.frame = rect
            navigationView.addNavigationItem(item: self.pinkieItem)
        }
    }

}
